plugins/org.komodo.modeshape.vdb/src/org/komodo/modeshape/visitor/DdlNodeVisitor.java

```diff
@@ -0,0 +1,971 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * See the COPYRIGHT.txt file distributed with this work for information
+ * regarding copyright ownership.  Some portions may be licensed
+ * to Red Hat, Inc. under one or more contributor license agreements.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ */
+package org.komodo.modeshape.visitor;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.Property;
+import javax.jcr.RepositoryException;
+import javax.jcr.Value;
+import javax.jcr.ValueFormatException;
+import javax.jcr.nodetype.NodeType;
+import org.komodo.modeshape.AbstractNodeVisitor;
+import org.komodo.modeshape.teiid.TeiidSqlNodeVisitor;
+import org.komodo.modeshape.teiid.parser.TeiidSQLConstants;
+import org.komodo.modeshape.teiid.parser.TeiidSQLConstants.NonReserved;
+import org.komodo.modeshape.teiid.parser.TeiidSQLConstants.Reserved;
+import org.komodo.spi.ddl.TeiidDDLConstants;
+import org.komodo.spi.metadata.MetadataNamespaces;
+import org.komodo.spi.runtime.version.TeiidVersion;
+import org.komodo.spi.type.DataTypeManager.DataTypeName;
+import org.komodo.spi.utils.KeyInValueHashMap;
+import org.komodo.utils.StringUtils;
+import org.modeshape.jcr.JcrLexicon;
+import org.modeshape.sequencer.ddl.StandardDdlLexicon;
+import org.modeshape.sequencer.ddl.dialect.teiid.TeiidDdlLexicon;
+
+/**
+ * Visitor that will walk a ddl node tree and convert it to
+ * the a string.
+ */
+public class DdlNodeVisitor extends AbstractNodeVisitor
```

Comment:
Thinking about this over the hols, I see the visitor as a specific implementation that would be accessed from api in KomodoObject, eg. ko.export(). There may well come a point when this visitor is pushed upstream to modeshape to complement the DdlSequencer.